betty/plugin/config.py

```diff
@@ -0,0 +1,169 @@
+        """
+        The human-readable short plugin label.
+        """
+        return self._plugin_label
+
+    @plugin_label.setter
+    def plugin_label(self, plugin_label: Localizable) -> None:
+        self._plugin_label = plugin_label
+        self._dispatch_change()
+
+    @property
+    def plugin_description(self) -> Localizable | None:
+        """
+        The human-readable long plugin description.
+        """
+        return self._plugin_label
+
+    @plugin_description.setter
+    def plugin_description(self, plugin_description: Localizable | None) -> None:
+        self._plugin_description = plugin_description
+        self._dispatch_change()
+
+    @override
+    def update(self, other: Self) -> None:
+        self._plugin_id = other.plugin_id
+        self._plugin_label = other.plugin_label
+        self._plugin_description = other.plugin_description
+
+    @override
+    @classmethod
+    def load(
+        cls,
+        dump: Dump,
+        configuration: Self | None = None,
+    ) -> Self:
+        asserter = Asserter()
+        plugin_id = asserter.assert_field(
+            RequiredField("plugin_id", Assertions(asserter.assert_str())),
+        )(dump)
+        configuration = cls(plugin_id, Str.plain(""))
+        asserter.assert_record(
+            Fields(
+                RequiredField("plugin_id"),
+                RequiredField(
+                    "plugin_label",
+                    # @todo These are not just strings.
+                    # @todo For both the label and the description,
+                    # @todo Make these new special Configuration objects (something like LocalizableConfiguration
+                    # @todo but specifically something that takes plain input strings
+                    # @todo - A single value is allowed, then no localization takes place
+                    # @todo - A mapping is allowed, where keys are locales and values are localized strings.
+                    # @todo
+                    # @todo
+                    Assertions(asserter.assert_str())
+                    | asserter.assert_setattr(configuration, "plugin_label"),
+                ),
+                OptionalField(
+                    "plugin_description",
+                    # @todo These are not just strings.
+                    Assertions(asserter.assert_str())
+                    | asserter.assert_setattr(configuration, "plugin_description"),
+                ),
+            )
+        )(dump)
+        return configuration
+
+    @override
+    def dump(self) -> VoidableDump:
+        dump: VoidableDictDump[VoidableDump] = {
+            "plugin_id": self.plugin_id,
+            # @todo How to dump our labels? We cannot use the Localizables directly
+            "plugin_label": self.plugin_label,
+            "plugin_description": void_none(self.plugin_description),
+        }
+
+        return minimize(dump)
+
+    @property
+    def plugin(self) -> type[Plugin]:
```

Comment:
Idea
- Plugins can be classes OR FACTORIES
- For generated plugins we then only use factories. No 'anonymous ' generated classes